dissect/update/__init__.py

```diff
@@ -0,0 +1,220 @@
+from __future__ import annotations
+
+import argparse
+import json
+import logging
+import os
+import subprocess
+import urllib.request
+from pathlib import Path
+from typing import Iterator
+
+from pip._vendor import tomli
+
+try:
+    import structlog
+
+    structlog.configure(
+        processors=[
+            structlog.contextvars.merge_contextvars,
+            structlog.processors.add_log_level,
+            structlog.processors.StackInfoRenderer(),
+            structlog.dev.set_exc_info,
+            structlog.processors.TimeStamper(fmt="%Y-%m-%d %H:%M:%S UTC", utc=True),
+            structlog.dev.ConsoleRenderer(),
+        ],
+        wrapper_class=structlog.make_filtering_bound_logger(logging.INFO),
+    )
+    log = structlog.get_logger()
+    HAS_STRUCTLOG = True
+
+except ImportError:
+    logging.basicConfig(level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s")
+    log = logging.getLogger(__name__)
+    HAS_STRUCTLOG = False
+
+
+PYPROJECT_FILE_PATHS = [
+    # wheels will have a toml file at dissect/update/pyproject.toml
+    str(Path(os.path.realpath(__file__)).parent) + "/pyproject.toml",
+    # tgz dist files will have a toml file at dissect/pyproject.toml
+    str(Path(os.path.realpath(__file__)).parent.parent) + "/pyproject.toml",
+    # git source repositories will have a toml file inside the repository root.
+    str(Path(os.path.realpath(__file__)).parent.parent.parent) + "/pyproject.toml",
+]
+
+PYPROJECT_ONLINE_URL = os.getenv(
+    "DISSECT_PYPROJECT_URL", "https://raw.githubusercontent.com/fox-it/dissect/main/pyproject.toml"
+)
+
+
+def _run(cmd: str, verbose: int) -> subprocess.CompletedProcess:
+    """Wrapper for subprocess run command."""
+    res = subprocess.run(cmd, shell=True, capture_output=True)
```

Comment:
Can you change calls to this to be `shell=False`? I don't think there's a need for it. The calls will look a bit "uglier" but unless it's absolutely necessary to use `shell=True` we should avoid it.